subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -383,6 +656,35 @@ int fp_storage_ak_save(const struct fp_account_key *account_key)
 		return err;
 	}
 
+	if (conn_ctx) {
+		__ASSERT_NO_MSG(IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_BOND_MANAGEMENT));
+
+		err = bond_find(&bond_idx, conn_ctx);
+		if (err) {
+			return err;
+		}
+
+		if (fp_bonds[bond_idx].bonded) {
+			bond = fp_bonds[bond_idx];
+
+			bond.nv.account_key_metadata_id = id;
+
+			err = bond_name_gen(bond_name, bond_idx);
+			if (err) {
+				return err;
+			}
+
+			err = settings_save_one(bond_name, &bond.nv, sizeof(bond.nv));
+			if (err) {
+				return err;
+			}
+
+			fp_bonds[bond_idx] = bond;
```

Comment:
You use this call sequence at least 3 times. Could you define a helper function to avoid duplication?

Reply:
If we opt for relying on `struct fp_bond *` instead of index, we could simplify internal APISs further. We could make a rollback copy and then update  record in an array directly through a pointer (on NVM error we would assign back the rollback copy). Storing function could then take only `struct fp_bond *` as an argument (index could be calculated internally as pointer diff; the function might even perform delete under the hood is data was zeroed in RAM).

Initially discussed the idea with @alstrzebonski and he was sceptical about it (mentioning inconsistency with other storage modules and introducing unnecessary changes here). Still, seemingly only this module does read->update->store, so using rollback mechanism here is justified in my opinion (other modules overwrite previous record with new data).